src/main/java/org/mastodon/views/trackscheme/display/ShowSelectedTracksActions.java

```diff
@@ -192,20 +192,50 @@ private void addEdgeTargets( final RefSet< TrackSchemeVertex > selected,
 		viewGraph.releaseRef( targetRef );
 	}
 
+	/**
+	 * @return the root nodes of the selected nodes, i.e. the nodes that are not
+	 * 	   descendants of other selected nodes. The order of the nodes in the returned
+	 * 	   list is from left to right as in the TrackScheme.
+	 */
 	private RefList< TrackSchemeVertex > filterRootNodes( final RefSet< TrackSchemeVertex > selectedVertices )
 	{
+		// Note: This method finds the root nodes of the nodes in the selectedVertices set.
+		// It does so by iterating over the graph in depth-first order and from left to right.
+		// Whenever the iteration hits a node that is in the selectedVertices set, it adds
+		// it to the returned list, and then skips all its child nodes. This guarantees that
```

Comment:
```suggestion
		// the root node from which the current iteration started to the returned list, and then skips all its child nodes. This guarantees that
```

Reply:
Hm. I got the impression that you misunderstood what the method does. I noticed that the method name was a little confusing. I changed it. How does the updated comment reads for you?

Reply:
Yes, there was a misunderstanding on my side. Comments and javadoc are good now.

Reply:
I removed the comments. The code is now simpler. It should be understandable.